### ISSUE TITLE
Add a ramp.getLegend method

### ISSUE THREAD

**Issue by davidmanzanares** [CartoDB/carto-vl#572]
By implementing ramp.eval we'll make legend-making much easier. However, I think we make it even easier if we provide some special method for the ramp class that returns, in the case of categories, the list of colors and their respective categories, and in the case of a linear ramp a list with the 1024 interpolated colors.

~Blocked for https://github.com/CartoDB/carto-vl/issues/570~

**Comment by Captain-Oski** [comment]
Following this feature-request tightly.

**Comment by elenatorro** [comment]
Blocked by https://github.com/CartoDB/carto-vl/issues/928

**Comment by elenatorro** [comment]
Closing in favor of #935
